File: front/src/api/suporte.ts
import api from './client'

export interface SuporteProtocolo {
  id: number
  id_cliente_servico: number
  descricao: string
  id_tipo_atendimento: number
  id_atendimento_status: number
  tipo_os?: number | null
  ativo: boolean
  data_hora_falha?: string | null
  protocolo_atendimento?: string | null
  bairro?: string | null
  clientes_afetados?: string | null
  informacoes_tecnicas?: any[] | null
  slot?: number | null
  pon?: number | null
  rota?: number | null
  cto?: number | null
  protocolo_id?: number | null
  responsavel_id?: number | null
  atendimento_suporte_id?: number | null
}

export interface SuporteListagemResponse {
  items: SuporteProtocolo[]
  total: number
  page: number
  per_page: number
  pages: number
}

export interface CreateSuportePayload {
  servico_id: number
  descricao?: string
  responsavel_id?: number | null
  id_tipo_atendimento: number
  id_atendimento_status?: number
  tipo_os?: number | null
  ativo?: boolean
  data_hora_falha?: string | null
  slot_pon?: string[]
  rotas?: number[]
  ctos?: number[]
  bairro?: string
  clientes_afetados?: string
}

export interface UpdateSuportePayload {
  descricao?: string
  responsavel_id?: number | null
  id_tipo_atendimento?: number
  id_atendimento_status?: number
  tipo_os?: number | null
  ativo?: boolean
  bairro?: string
  clientes_afetados?: string
}

export async function listarSuporte(params?: Record<string, any>): Promise<SuporteProtocolo[]> {
  const { data } = await api.get<SuporteListagemResponse>('/suporte/', { params })
  return data.items
}

export async function getSuporte(id: number): Promise<SuporteProtocolo> {
  const { data } = await api.get<SuporteProtocolo>(`/suporte/${id}`)
  return data
}

export async function criarSuporte(payload: CreateSuportePayload): Promise<{ status: string; message: string; id?: number }> {
  const { data } = await api.post('/suporte/', payload)
  return data
}

export async function atualizarSuporte(id: number, payload: UpdateSuportePayload): Promise<{ status: string; message: string }> {
  const { data } = await api.put(`/suporte/${id}`, payload)
  return data
}

export async function desativarSuporte(id: number): Promise<{ status: string; message: string }> {
  const { data } = await api.delete(`/suporte/${id}`)
  return data
}

export async function atualizarStatusSuporte(id: number, id_atendimento_status: number): Promise<{ status: string; message: string }> {
  const { data } = await api.patch(`/suporte/${id}/status`, { id_atendimento_status })
  return data
}

File: back/app/schemas/suporte.py
from pydantic import BaseModel, ConfigDict, Field
from datetime import datetime
from typing import Optional


class SuporteProtocoloCreate(BaseModel):
    servico_id: int
    descricao: Optional[str] = ""
    responsavel_id: Optional[int] = None
    id_tipo_atendimento: int = Field(..., description="Tipo de atendimento FTTH")
    id_atendimento_status: int = 1
    tipo_os: Optional[int] = None
    ativo: bool = True
    data_hora_falha: Optional[str] = None
    slot_pon: list[str] = []
    rotas: list[int] = []
    ctos: list[int] = []
    bairro: Optional[str] = ""
    clientes_afetados: Optional[str] = ""


class SuporteProtocoloUpdate(BaseModel):
    descricao: Optional[str] = None
    responsavel_id: Optional[int] = None
    id_tipo_atendimento: Optional[int] = None
    id_atendimento_status: Optional[int] = None
    tipo_os: Optional[int] = None
    ativo: Optional[bool] = None
    bairro: Optional[str] = None
    clientes_afetados: Optional[str] = None


class SuporteStatusUpdate(BaseModel):
    id_atendimento_status: int = Field(..., ge=1, le=5, description="1=Pendente, 2=Análise, 22=Andamento, 3=Resolvido, 4=Fechado")


class SuporteDetalhesResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: int
    protocolo_id: Optional[int] = None
    responsavel_id: Optional[int] = None
    atendimento_suporte_id: Optional[int] = None
    id_cliente_servico: int
    descricao: str
    id_tipo_atendimento: int
    id_atendimento_status: int
    tipo_os: Optional[int] = None
    ativo: bool
    data_hora_falha: Optional[datetime] = None
    protocolo_atendimento: Optional[str] = None
    slot: Optional[int] = None
    pon: Optional[int] = None
    rota: Optional[int] = None
    cto: Optional[int] = None
    bairro: Optional[str] = None
    clientes_afetados: Optional[str] = None
    informacoes_tecnicas: Optional[list] = None


class SuporteListagemItem(BaseModel):
    id: int
    id_cliente_servico: int
    id_tipo_atendimento: int
    id_atendimento_status: int
    tipo_os: Optional[int] = None
    ativo: bool
    descricao: str
    data_hora_falha: Optional[datetime] = None
    bairro: Optional[str] = None


class SuporteListagemResponse(BaseModel):
    items: list[SuporteListagemItem]
    total: int
    page: int
    per_page: int
    pages: int
